crates/bifrost/src/loglets/memory_loglet.rs

```diff
@@ -300,14 +326,14 @@ impl LogletBase for MemoryLoglet {
         Ok(first_offset)
     }
 
-    async fn find_tail(&self) -> Result<Option<LogletOffset>> {
-        let log = self.log.lock().unwrap();
-        if log.is_empty() {
-            Ok(None)
+    async fn find_tail(&self) -> Result<TailState<LogletOffset>> {
+        let committed = LogletOffset(self.last_committed_offset.load(Ordering::Acquire)).next();
+        let sealed = self.sealed.load(Ordering::Acquire);
```

Comment:
Why are we using `Ordering::Acquire` for `last_committed_offset` and `sealed` here?

Reply:
Probably historical reasons, this is completely unnecessary.